lighthouse-core/report/html/renderer/details-renderer.js

```diff
@@ -536,24 +536,50 @@ class DetailsRenderer {
       return null;
     }
 
+    /** @param {HTMLElement} el */
+    function addDevToolsData(el) {
+      el.classList.add('lh-source-location');
+      el.setAttribute('data-source-url', item.url);
+      // DevTools expects zero-indexed lines.
+      el.setAttribute('data-source-line', String(item.line));
+      el.setAttribute('data-source-column', String(item.column));
+      return el;
+    }
+
     // Lines are shown as one-indexed.
-    const line = item.line + 1;
-    const column = item.column;
+    const line = (item.original ? item.original.line : item.line) + 1;
+    const column = item.original ? item.original.column : item.column;
 
-    let element;
+    // First two cases: url is from network.
     if (item.urlProvider === 'network') {
-      element = this.renderTextURL(item.url);
-      this._dom.find('a', element).textContent += `:${line}:${column}`;
+      let element;
+      if (item.original) {
+        element = this._renderLink({
+          url: item.url,
+          text: `${item.original.file}:${line}:${column}`,
+        });
+        element.title = `${item.url}:${line}:${column}`;
+      } else {
+        element = this.renderTextURL(item.url);
+        this._dom.find('a', element).textContent += `:${line}:${column}`;
+      }
+
+      return addDevToolsData(element);
+    }
+
+    // Two cases remain:
+    // 1) urlProvider === 'comment' && item.original
+    // 2) urlProvider === 'comment' && !item.original
```

Comment:
```suggestion
    // 1) urlProvider === 'comment' && item.original - Show the source filename in text, source mapped URL in tooltip.
    // 2) urlProvider === 'comment' && !item.original - Show the source mapped URL.
```

Or whatever language you feel makes it the clearest what we're *supposed* to do in each situation?